src/main/kotlin/org/rust/lang/refactoring/RustNameSuggestions.kt

```diff
@@ -0,0 +1,101 @@
+package org.rust.lang.refactoring
+
+import com.intellij.openapi.project.Project
+import com.intellij.psi.PsiElement
+import com.intellij.psi.codeStyle.NameUtil
+import com.intellij.psi.util.PsiTreeUtil
+import org.rust.ide.navigation.goto.RustSymbolNavigationContributor
+import org.rust.lang.core.psi.*
+import org.rust.lang.core.psi.util.childOfType
+import org.rust.lang.core.psi.util.parentOfType
+import org.rust.lang.core.types.util.resolvedType
+import java.util.*
+
+/**
+ * This suggests names for an expression about to be bound to a local variable
+ *
+ * If the type is resolved suggest the first letter of the type name.
+ * If its an argument to a function call, suggest the name of the argument in the function definition.
+ * If its a function call suggest the name of the function and the enclosing name space (if any).
+ * If a name is already bound in the local scope do not suggest it.
+ */
+fun PsiElement.suggestNames(): LinkedHashSet<String> {
+    val names = LinkedHashSet<String>()
+    nameForType(this)?.let { names.addAll(RustNameUtil(it)) }
+
+    val foundNames = when {
+        this.isArgument() -> RustNameUtil(nameForArgument())
+        this is RustCallExprElement -> nameForCall(this).flatMap(::RustNameUtil)
+        else -> emptyList()
+    }
+
+    names.addAll(foundNames)
+
+    val usedNames = findNamesInLocalScope(this)
+    names.removeAll(usedNames)
+
+    return names
+}
+
+private fun nameForType(expr: PsiElement): String? {
+    val typeString = (expr as? RustExprElement)?.resolvedType.toString()
+    //doesn't really seem to do that much.
+    if (typeString != "unknown" && typeString != "null") {
```

Comment:
Let's be more specific here and return `null` if `type is RustUnknownType || type == null`